factories/settingsValidatorFactory.js

```diff
@@ -94,6 +94,9 @@
         else if ( self.searchEngine === 'searchapi' ) {
           return doc.doc;
         }
+        else if ( self.searchEngine === 'algolia' ) {
```

Comment:
lovely!  Someday there is probably a better way then an ever expanding list of if statements!

Reply:
We could have a map of functions and use property lookup to invoke the function. Maybe something like:

```js
      var docSourceConfiguration = {
        solr: function(doc) {
          return doc.doc;
        },
        es: function (doc) {
          return doc.doc._source;
        },
        os: function (doc) {
          return doc.doc._source;
        },
        vectara: function(doc) {
          // Vectara returns doc properties in a metadata array of objects containing 'name' + 'value pairs
          const fieldsFromDocumentMetadata = doc.doc.metadata.reduce(function(map, obj) {
            map[obj.name] = obj.value;
            return map;
          }, {});
          return Object.assign({}, {
            'id': doc.doc.id
          }, fieldsFromDocumentMetadata);
        },
        searchapi: function(doc) {
          return doc.doc;
        },
        algolia: function(doc) {
          return doc.doc;
        },
      }

      function sourceDoc(doc) {
        if (docSourceConfiguration.hasOwnProperty(self.searchEngine)) {
          return docSourceConfiguration[self.searchEngine]();
        } else {
          console.error('Need to determine how to source a doc for this search engine ' + self.searchEngine);
        }
      }
```

Reply:
yeah, lets think about it in another PR ;-)